### DISCUSSION TITLE
Project structure

### DISCUSSION THREAD

**Discussion by bastienfaivre** [BastienFaivre/ExchangeHub#2] [Ideas]
Discuss the project structure

**Comment by bastienfaivre** [comment]
As discussed today, we defined a basic structure of the project as follow:

```
ExchangeHub/
├─ public/
│  ├─ ...
├─ src/
│  ├─ helpers/
│  │  ├─ ...
│  ├─ home/
│  │  ├─ homePresenter.js
│  │  ├─ homeView.js
│  ├─ mainContent/
│  │  ├─ Courses/
│  │  │  ├─ coursePresenter.js
│  │  │  ├─ courseView.js
│  │  │  ├─ searchPresenter.js
│  │  │  ├─ searchView.js
│  │  ├─ Lifestyle/
│  │  │  ├─ searchPresenter.js
│  │  │  ├─ searchView.js
│  │  │  ├─ tipPresenter.js
│  │  │  ├─ tipView.js
│  │  ├─ Profile/
│  │  │  ├─ coursesPresenter.js
│  │  │  ├─ coursesView.js
│  │  │  ├─ informationPresenter.js
│  │  │  ├─ informationView.js
│  │  │  ├─ lifestylePresenter.js
│  │  │  ├─ lifestyleView.js
│  │  ├─ Students/
│  │  │  ├─ searchPresenter.js
│  │  │  ├─ searchView.js
│  │  │  ├─ studentPresenter.js
│  │  │  ├─ studentView.js
│  ├─ navigation/
│  │  ├─ navigationPresenter.js
│  │  ├─ navigationView.js
│  ├─ sideBar/
│  │  ├─ sideBarPresenter.js
│  │  ├─ sideBarView.js
│  ├─ index.html
│  ├─ index.js
```
This is of course just a draft structure that will be update as we dive further into the project. Feel free to comment if there is something that needs to be modified or add.

**Reply by phamelink** [reply]
Sounds good, one slight thing I would add is eliminate the mainContent folder. When working with components in React you very often have to import other components to your project, and having unnecessary nested files can make it harder to organise and maintain in the long run (trust me ahah).
But it's a small detail, I think it looks good!

**Comment by bastienfaivre** [comment]
We can notice that we have 3 times a search instance. As we develop the website, it would be interesting to see if building a single search instance is possible.
